niceman/tests/test_utils.py

```diff
@@ -435,12 +422,40 @@ def test_hashable_dict():
     assert(key_c not in d)
 
 
-def test_batch_process_list():
-    # Let's sum the numbers from 1 to 100 in different batches
-    l = list(range(1,101))
-    f = lambda b, p: sum(b) + p
-    s = sum(l)  # I know, it should be 5050 :)
-    assert(batch_process_list(f, l, 200, 0) == s)
-    assert(batch_process_list(f, l, 50, 0) == s)
-    assert(batch_process_list(f, l, 10, 0) == s)
-    assert(batch_process_list(f, l, 1, 0) == s)
+def test_execute_command_batch():
+    # Create a dummy session that can possibly raise a ValueError
+    class DummySession(object):
+        def execute_command(self, cmd):
+            if cmd[0] == "ValueError":
+                raise ValueError
+            else:
+                return (len(cmd), None)
+    session = DummySession()
+    # First let's do a simple test to count the args
+    args = list(map(str, range(1, 101)))
+    cmd_gen = execute_command_batch(session, [], args, None)
+    for (out, _, _) in cmd_gen:
+        assert out == 100
+    # Now let's raise an exception but not list it as handled
+    cmd_gen = execute_command_batch(session, ["ValueError"], args, None)
+    with pytest.raises(ValueError):
+        for (_, _, _) in cmd_gen:
+            pass
```

Comment:
This for loop could be reduced to `next(cmd_gen)`.

Reply:
good call! it would indeed should be sufficient here!
but in general: `next` would give just a single element, while the loop would go through all. so if it was desired,  `list` instead of `next` should be used ;)

Reply:
> but in general

Sure, we only know `next` is sufficient in this particular case
because we constructed the one item list two lines above.  But from a
readability/intention standpoint, I agree it'd be better to use
`list`.

Reply:
Done.